api/incubator/src/main/java/io/opentelemetry/api/incubator/common/ExtendedAttributeType.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.incubator.common;
+
+/**
+ * An enum that represents all the possible value types for an {@link ExtendedAttributeKey} and
+ * hence the types of values that are allowed for {@link ExtendedAttributes}.
+ *
+ * <p>This is a superset of {@link io.opentelemetry.api.common.AttributeType},
+ */
+public enum ExtendedAttributeType {
+  // Types copied AttributeType
+  STRING,
+  BOOLEAN,
+  LONG,
+  DOUBLE,
+  STRING_ARRAY,
+  BOOLEAN_ARRAY,
+  LONG_ARRAY,
+  DOUBLE_ARRAY,
+  // Extended types unique to ExtendedAttributes
+  EXTENDED_ATTRIBUTES;
```

Comment:
- Dropped arrays of maps for now, per this comment: https://github.com/open-telemetry/opentelemetry-java/pull/6983#discussion_r1927386922
- Renamed from MAP -> EXTENDED_ATTRIBUTES per this comment: https://github.com/open-telemetry/opentelemetry-java/pull/6983#discussion_r1927383394

Should we consider adding support for byte arrays?

Reply:
I don't think it has to be now. Users will base64 encode or json stuff anyway. 🤷🏻

Reply:
Yeah users will always have that ability, but having direct representation for a type a signal that there is a recommended way of representing the data. This encourages more consistency than being opinionated, which will produce a bit of a free for all in terms of what instrumentations produce and what backends receive.

Still, I'm happy to punt on these questions for now and let semantic conventions develop more guidance.